src/InputValidationMiddleware.php

```diff
@@ -0,0 +1,59 @@
+<?php
+namespace Aws;
+
+use Aws\Api\Service;
+
+/**
+ * Validates the required input parameters of commands are non empty
```

Comment:
After some discussion, an approach to limit the scope to specific members may be the better approach, in case new operations are released with required members that may not necessarily conform to this validation. The middleware should be able to take in an argument containing relevant members.